Fixed site setup correctly naming the default Tier

allouis opened the pull request:
no-issue

Since the Free tier was added, the first Tier was not the default one. This updates the site setup to check for the slug explicitly

allouis commented on core/server/services/auth/setup.js:
```diff
@@ -110,14 +110,15 @@ async function doProduct(data, productsAPI) {
     }
     try {
-        const page = await productsAPI.browse({limit: 1});
+        const page = await productsAPI.browse({limit: 'all'});
```
Need to fetch all, because we cannot assume first one is free - generally there will be 2

allouis commented on core/server/services/auth/setup.js:
```diff
@@ -110,14 +110,15 @@ async function doProduct(data, productsAPI) {
         }
 
-        productsAPI.edit({products: [{name: blogTitle.trim()}]}, {context: context.context, id: product.id});
+        await productsAPI.edit({products: [{name: blogTitle.trim()}]}, {context: context.context, id: product.id});
```
We await it so that we can catch errors

naz commented on test/utils/e2e-framework-mock-manager.js:
```diff
@@ -13,6 +13,14 @@ let emailCount = 0;
 const mailService = require('../../core/server/services/mail/index');
 const labs = require('../../core/shared/labs');
 
+const disableStripe = async () => {
```
nice :ok_hand::ok_hand::ok_hand:

allouis committed:
Added disableStripe method to mock manager

Some tests need to run without Stripe connected/enabled, which is the
default at the moment. This method will disconnect stripe for the
lifecycle of the API agent.

allouis committed:
Fixed new sites using "Default Product" Tier name

https://github.com/TryGhost/Team/issues/1289

Since we added the free Tier fixture, this was attempting to fix the
free Tier, rather than the default one.

allouis committed:
Renamed Stripe Products from "Default Product"

refs https://github.com/TryGhost/Team/issues/1289

Due to a bug in site setup some Tiers were named "Default Product" and
so the corresponding Stripe Product was also named this. This adds a
Stripe migration to rename any Stripe Products from "Default Product" to
the site title.
